src/main/java/masecla/reddit4j/objects/RedditPost.java

```diff
@@ -140,71 +141,71 @@ public class RedditPost extends RedditThing implements RedditPostActions {
     /**
      * full URL to the thumbnail for this link; "self" if this is a self post; "image" if this is a link to an image but has no thumbnail; "default" if a thumbnail is not available
      */
-    String thumbnail;
+    private String thumbnail;
 
     /**
      * the title of the link. may contain newlines for some reason
      */
-    String title;
+    private String title;
 
     /**
      * the link of this post. the permalink if this is a self-post
      */
-    String url;
+    private String url;
 
     /**
      * Indicates if link has been edited. Will be the edit timestamp if the link has been edited and return false otherwise. https://github.com/reddit/reddit/issues/581
      * TODO: The API can return a long or a boolean, needs custom deserialization to handle it
      */
-    //long edited;
+    //private long edited;
 
     /**
      * to allow determining whether they have been distinguished by moderators/admins. null = not distinguished. moderator = the green [M]. admin = the red [A]. special = various other special distinguishes http://bit.ly/ZYI47B
      */
-    String distinguished;
+    private String distinguished;
 
     /**
      * true if the post is set as the sticky in its subreddit.
      */
-    boolean stickied;
+    private boolean stickied;
 
     @SerializedName("author_fullname")
 	private String authorFullname;
 
-    String modReasonTitle;
-    int gilded;
-    List<Object> linkFlairRichtext;
-    String subredditNamePrefixed;
-    int pwls;
+    private String modReasonTitle;
+    private int gilded;
+    private List<Object> linkFlairRichtext;
+    private String subredditNamePrefixed;
+    private int pwls;
 
     /**
      * The number of upvotes. (includes own).
      *
      * https://github.com/reddit-archive/reddit/wiki/JSON#votable-implementation
      */
-    int ups;
+    private int ups;
 
     /**
      * The number of downvotes. (includes own).
      *
      * https://github.com/reddit-archive/reddit/wiki/JSON#votable-implementation
      */
-    int downs;
+    private int downs;
 
     /**
      * true if thing is liked by the user, false if thing is disliked, null if the user has not voted or you are not logged in.
      * Certain languages such as Java may need to use a boolean wrapper that supports null assignment.
      *
      * https://github.com/reddit-archive/reddit/wiki/JSON#votable-implementation
      */
-    Boolean liked;
+    private Boolean liked;
```

Comment:
Shouldn't this simply be a bool?

Reply:
Documentation linked mentions it can be null

Reply:
We can't use the primitive bool as the api uses null for Posts that are neither upvoted or downvoted